src/ceph-daemon/ceph-daemon

```
@@ -436,6 +503,81 @@ def create_daemon_dirs(fsid, daemon_type, daemon_id, uid, gid,
+
+        config = get_parm(args.config_json)
+        if not config:
+            raise Error("Prometheus deployment requires config-json to be set")
+
+        prometheus_yml = dict()
+
+        if "prometheus.yml" in config:
+            # trust the provided version is appropriate
+            prometheus_yml = config.get('prometheus.yml')
+            logger.debug("Using supplied prometheus.yml definition")
+        else:
+            if "mgrs" in config:
+                prometheus_yml = Monitoring.components['prometheus']['config']['prometheus.yml']
+                if isinstance(config['mgrs'], list):
+                    targets=[]
+                    for mgr in config['mgrs']:
+                        targets.append('{}:9283'.format(mgr))
+                    prometheus_yml['scrape_configs'].append({
+                            "job_name": "ceph",
+                            "static_configs": [
+                                { "targets": targets,
+                                  "labels": {
+                                    "instance": "ceph_cluster"
+                                    }
+                                }
+                            ],
+                        })
+                    logger.debug("Using default prometheus settings with supplied list of ceph mgrs")
+                else:
+                    raise Error("config-json supplied with invalid mgrs setting - must be a list")
+            else:
+                raise Error("config-json must either specify a valid prometheus.yml, or a mgrs list")
+
+        # create configs
+        with open(os.path.join(data_dir_root,'etc/prometheus/prometheus.yml'), 'w') as f:
+            os.fchown(f.fileno(), uid, gid)
+            os.fchmod(f.fileno(), 0o600)
+            f.write(yaml.safe_dump(prometheus_yml, explicit_start=True, indent=2))
+
+def get_parm(option):
+    """Read requested arg, and return as a dict"""
+
+    if not option:
+        return None
```

:-1: for silently returning `None`. Can we raise an exception here?

---

We return none, and in the caller decide whether none is a problem or not. in this case it is and we raise an exception.